### DISCUSSION TITLE
How to run adjoint optimization locally?

### DISCUSSION THREAD

**Discussion by triplestone0818** [flexcompute/tidy3d#1574] [Q&A] [answered]
Due to the limited 400 structures in the remote cloud server, is there any else way to simulate more than 400 structures with tidy3d? The method now i want to do is run adjoint optimization in my local server, but i don't know whether it is practical or not.

**Comment by momchil-flex** [comment, marked as answer]
Yeah, actually you should be able to do `from tidy3d.plugins.adjoint.web import run_local`, use that instead of the usual adjoint run call, and comment out the maximum structure validator in your local code.

**Reply by triplestone0818** [reply]
Thank you, this method seems to work, now there is no limit for the number of structures, so that i can simulate a large diameter metalens in my local server.

**Comment by tylerflex** [comment]
@momchil-flex I don't think that will work because our server will error due to this validator right?

**Reply by momchil-flex** [reply]
No, `run_local` only submits a regular Simulation, not a JaxSimulation.

**Comment by tylerflex** [comment]
One approach could be to make these "post-init" validators and only call them when we use the remote `tda.web.run()`? (ie don't validate this on `local_run`?

**Reply by triplestone0818** [reply]
Thank you your answer!

**Comment by tylerflex** [comment]
Closing this, @triplestone0818 in a future release, we will make this a bit more convenient through #1575 (so you won't need to comment out the validators manually)
